service_vertices: [[10, 10], [10, -10], [-10, 10], [-10, -10]]